Pinebook Pro | Add initial support

nonnorm opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Ready

**Reference**: https://github.com/MichaIng/DietPi/issues/3932

**Commit list/description**:
<!--
- DietPi-Config | Add "Fan control" option to "Performance Options"
![Screenshot](https://xxx.github.com/images/xxx.png)
- DietPi-Config | Add "Fan control" support for Odroid C2
- DietPi-Config | Syntax fix
-->
- DietPi-PREP | Add PBP as supported device
- DietPi-PREP | Change PBP ID to be in line with other Pine64 products

nonnorm committed:
v7.2
+ DietPi-PREP | Add PBP as supported device

nonnorm committed:
v7.2
+ DietPi-Software | Change PBP ID to be in line with other Pine64 products

nonnorm committed:
v7.2
+ DietPi-PREP | Move PBP to Limited GPU acceleration

nonnorm committed:
v7.2
+ DietPi-Imager | Fix DietPi-Imager

nonnorm committed:
v7.2
+ DietPi-Imager | Fixes

MichaIng committed:
v7.2

+ DietPi-Obtain_HW_model | Add Pinebook Pro name and SoC ID (RK3399)

MichaIng commented on .meta/dietpi-imager:
```diff
@@ -87,7 +87,14 @@
+			done
+
+			# Sort array
+			G_WHIP_MENU_ARRAY=($(tr ' ' '\n' <<< "${G_WHIP_MENU_ARRAY[@]}" | mawk '!seen[$0]++' | tr '\n' ' '))
```
Indeed, the sort is misplaced there, removing all but one "ext4" line and breaking the intended array composition. May I suggest a simpler solution:
```
mapfile -t G_WHIP_MENU_ARRAY < <(mawk -v root="$(lsblk -npo PKNAME "$G_ROOTFS_DEV")" '{if ( $1!=root && $2=="ext4" ) print $1"\n"$1"_details"}' < <(lsblk -rnpo PKNAME,FSTYPE) | sort -u)
```
When PKNAME = $1 is a duplicate, which is what we want to sort out, then <PKNAME>_details is a duplicate as well, so both are sorted out together. The second field/line is a only a placeholder to replace with more helpful information, as in bash one cannot insert array entries at a specific index and having all later entries moved one index upwards.

Probably a separate temporary array would be cleaner that contains only PKNAME entries, removes duplicates, and the second loop then creates G_WHIP_MENU_ARRAY from scratch from temporary array + details. This placeholder method looks a bit hacky 😄.

nonnorm replied:
That's much better, I was trying to sort the array after adding the extra details.

MichaIng committed:
v7.2

+ DietPi-Survey_report | Add support for Pinebook Pro

MichaIng committed:
v7.2

+ DietPi-PREP | Assure that rng-tools5 is installed on Pinebook Pro as well, as RK3399 ships with functional hardware random generator

nonnorm committed:
v7.2
+ DietPi-Imager | Change G_WHIP_MENU_ARRAY compisition

nonnorm committed:
Merge branch 'pinebook-pro' of https://github.com/MichaIng/DietPi into pinebook-pro

MichaIng committed:
v7.2

+ DietPi-Imager | Tiny: Re-add space after ":" in right-hand menu text, to align with other cases

MichaIng committed:
v7.2

+ CHANGELOG | Pinebook Pro | Initial hardware identifier (ID: 46) and support for this device has been added to DietPi.

MichaIng committed:
Merge branch 'dev' into pinebook-pro
